Core/src/EventData/PrintParameters.cpp

```diff
@@ -0,0 +1,59 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2016-2020 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "Acts/EventData/detail/PrintParameters.hpp"
+
+#include <ostream>
+
+#include "Acts/Surfaces/Surface.hpp"
+
+void Acts::detail::printBoundParameters(std::ostream& os,
+                                        const Acts::Surface& surface,
+                                        const Acts::BoundVector& params,
+                                        const Acts::BoundSymMatrix* cov) {
```

Comment:
Why did you use a pointer instead of the `std::optional`?

Reply:
So this can be used for cases where the covariance is not stored as via `std::optional` and I need to include one header less.

I do not want to use `std::optional<BoundSymMatrix>` to avoid copies of large matrices. Using `const std::optional<BoundSymMatrix>&` has the same issue if the matrix is not already stored as a `std::optional`. I could use `std::optional<const BoundSymMatrix&>` but I think that is not supported by `std::optional`.

Reply:
I see. One could argue that the parameters have to fulfill the concept and therefore have to provide a getter for an optional covariance but since the internal storage is not fixed by it, the argument is a bit vague and I'm fine with it. The `optional<type&>` is possible in boost but not for `std::optional`.